### ISSUE TITLE
Question: Use action / hooks / filter?

### ISSUE THREAD

**Issue by martincodes-de** [phpsword/sword-bundle#3]
Hey, is it posible to use action / hooks and filters? To, as example, fire some from the symfony part of the application the to the wordpress part?

**Comment by mechanicalgux** [comment]
Yes you can use all WordPress functions in Symfony as long as they are called after it's loaded. First you have to [enable your theme](https://getsword.com/guide/tutorial/first-steps.html) to make it Symfony compatible, then you create [WordPress-aware services](https://getsword.com/guide/tutorial/services.html). Those are able to call WordPress functions and hooks. Any service called by a `WordpressService` is able to make use of WordPress functions and hooks.

**Comment by williarin** [comment]
Symfony and WordPress are merged together, so yes you can call WordPress actions and filters from Symfony.

**Comment by loicrey** [comment]
Is it possible to do this before Wordpress initialized (which initialize after Symfony) ? 
I need to call some filters with `apply_filters` before `initialize_services()` is called in my child theme. I can't trigger Wordpress Hooks to achieve this because I need to call it when my controller is executed

**Comment by williarin** [comment]
What do you think of this solution (not tested), assuming you're using a custom Symfony controller that triggers the Wordpress loader:

1. create a Symfony custom event `MyEvent extends Event`
2. create a Wordpress service `MyService extends AbstractWordpressService`, and in the constructor your inject `EventDispatcherInterface` and immediately listen to `MyEvent`. In the listener you store the event data in a property of `MyService`
3. in your Symfony controller, you dispatch `MyEvent` with the data you need, before Wordpress initialization. At this point your data is stored in `MyService` but Wordpress isn't initalized yet
4. call `apply_filters` in `MyService->initialize()`, using the data you previously stored

**Comment by loicrey** [comment]
Hi William.

It's a controller who renders a Twig, so controller is not triggering the Wordpress loader I think

**Comment by williarin** [comment]
Hi Loïc,

From what I understand, you want to make Wordpress render some html or some output in your controller using `apply_filters`, then print it in your Twig template.

Some possibilities:
* create a Wordpress API endpoint, then call it with `file_get_contents()` before rendering the view
* render nothing then call an ajax request to Wordpress that will return the rendered output

Not sure how to do it otherwise as Wordpress can't be unloaded... So you must find a way to isolate the rendering process.
